src/Catalog/Registration/RegistrationCollector.cs

```diff
@@ -88,6 +90,8 @@ private string GetKey(JToken item)
         {
             var tasks = new List<Task>();
 
+            // TODO: this is where processing of a package ends. 
```

Comment:
This is where the Registration blobs are saved. Perhaps use a Task.ContinueWith to complete the measurement?